### PR TITLE
Minor: Improve ExecutionMode documentation

### PR DESCRIPTION
**Pull request by alamb**
N/A

## Rationale for this change

While working with @ozankabak  and @berkaysynnada on https://github.com/apache/datafusion/pull/12174.  I found myself referring several times to the `ExecutionMode` documentation and had to re-read it a few times 

## What changes are included in this PR?
Try to improve the docs

### CODE REVIEW

File: datafusion/physical-plan/src/execution_plan.rs [hunk 1]
```diff
@@ -482,24 +482,41 @@ impl ExecutionPlanProperties for &dyn ExecutionPlan {
...
     Unbounded,
-    /// Represents the mode where some of the operator's input stream(s) are
-    /// unbounded; however, the operator cannot generate streaming results from
-    /// these streaming inputs. In this case, the execution mode will be pipeline
-    /// breaking, e.g. the operator requires unbounded memory to generate results.
+    /// Some of the operator's input stream(s) are unbounded, but the  operator
+    /// cannot generate streaming results from these streaming inputs.
+    ///
+    /// In this case, the execution mode will be pipeline breaking, e.g. the
+    /// operator requires unbounded memory to generate results.
```
Comment (ozankabak) [inline datafusion/physical-plan/src/execution_plan.rs:519]:
```suggestion
    /// Some of the operator's input stream(s) are unbounded, but the operator
    /// cannot generate streaming results from these streaming inputs.
    ///
    /// In this case, the execution mode will be pipeline breaking, e.g. the
    /// operator requires unbounded memory to generate results. This
    /// information is used by the planner when performing sanity checks
    /// on plans processings unbounded data sources.
```

**Review by ozankabak — APPROVED** [review]
LGTM, added some more info to docstrings

File: datafusion/physical-plan/src/execution_plan.rs [hunk 2]
```diff
@@ -482,24 +482,41 @@ impl ExecutionPlanProperties for &dyn ExecutionPlan {
...
+///
+/// The mode of the execution plan is determined by the mode of its input
+/// execution plans and the details of the operator itself. For example, a
+/// `FilterExec` operator will have the same execution mode as its input, but a
+/// `SortExec` operator may have a different execution mode than its input,
+/// depending on how the input stream is sorted.
+///
+/// There are three possible execution modes: `Bounded`, `Unbounded` and
+/// `PipelineBreaking`.
+# [derive(Clone, Copy, PartialEq, Debug)]
```
Comment (ozankabak) [inline datafusion/physical-plan/src/execution_plan.rs:496]:
```suggestion
#[derive(Clone, Copy, PartialEq, Debug)]
```
Whitespace creates a fmt error

Comment (alamb) [reply datafusion/physical-plan/src/execution_plan.rs:496]: Thank you -- sorry about that

**Comment by alamb** [comment]
Thanks again @ozankabak

**Pull request merged**